api/src/main/java/org/neo4j/ogm/config/Configuration.java

```diff
@@ -97,22 +97,8 @@
         this.useNativeTypes = builder.useNativeTypes;
         this.basePackages = builder.basePackages;
 
-        if (this.uri != null) {
-            URI parsedUri = null;
-            try {
-                parsedUri = new URI(this.uri);
-            } catch (URISyntaxException e) {
-                throw new RuntimeException("Could not configure supplied URI in Configuration", e);
-            }
-            String userInfo = parsedUri.getUserInfo();
-            if (userInfo != null) {
-                String[] userPass = userInfo.split(":");
-                credentials = new UsernamePasswordCredentials(userPass[0], userPass[1]);
-                this.uri = parsedUri.toString().replace(parsedUri.getUserInfo() + "@", "");
-            }
-            if (getDriverClassName() == null) {
-                this.driverName = Drivers.getDriverFor(parsedUri.getScheme()).driverClassName();
-            }
+        if (this.uri != null || (this.uris != null && uris.length >= 1)) {
```

Comment:
Can we replace this check wit a call to `getSingleURI`, save the returned value, check for null and pass that value onto `parseAndSetParametersFromURI` instead of calling `getSingleURI` there?